src/server/analytic_services/prometheus_reporting.js

```diff
@@ -69,6 +70,21 @@ async function start_server(
         return;
     }
     const metrics_request_handler = async (req, res) => {
+        // TODO: This is a temporary condition to avoid the token authentication for metrics, 
+        // Need to add authentication for NSFS NC also after confirming flow with the Scale team
+        if (!process.env.NC_NSFS_NO_DB_ENV) {
```

Comment:
What if we make the authentication itself behind a config which is `true` by default but can be disabled via config (and consequently via env)?

Note: I lack all the context here so if this is meaningless please ignore.

Reply:
I still believe that it could be beneficial if we have a config dedicated to toggling on and off this feature.